share/get_proof_test.go

```diff
@@ -0,0 +1,88 @@
+package share
+
+import (
+	"context"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	mdutils "github.com/ipfs/go-merkledag/test"
+	"github.com/minio/sha256-simd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/share/ipld"
+)
+
+func TestGetSharesWithProofsByNamespace(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	bServ := mdutils.Bserv()
+
+	var tests = []struct {
+		rawData []Share
+	}{
+		{rawData: RandShares(t, 4)},
+		{rawData: RandShares(t, 16)},
+		{rawData: RandShares(t, 64)},
+	}
+
+	for i, tt := range tests {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			rand.Seed(time.Now().UnixNano())
+			// choose random range in shares f
+			from := rand.Intn(len(tt.rawData) - 1)
+			to := rand.Intn(len(tt.rawData) - 1)
+
+			//
+			expected := tt.rawData[from]
+			nID := expected[:NamespaceSize]
+
+			if to < from {
+				tmp := from
+				from, to = to, tmp
+			}
+
+			// change rawData to contain several shares with same nID
+			for i := from; i <= to; i++ {
+				tt.rawData[i] = expected
```

Comment:
could you maybe only write the nID instead of using the same share?

Reply:
i see that it would make the share validation check more complicated though :/ but whatever you think is a better test, ill trust

Reply:
Yeah, it gets more complicated.